src/store/plugins/router.js

```diff
@@ -50,15 +50,32 @@ export default store => {
   })
 
   router.beforeEach(async (to, from, next) => {
+    for (let m of from.matched) {
+      if (m.meta.onLeave) {
+        console.log('onLeave', m.meta.onLeave)
+        store.dispatch(m.meta.onLeave)
```

Comment:
Maybe it makes a difference running it an `afterEach` handler? Would make the sequence of things clearer to me. No specific reason, but my senses tell me the ordering could confuse things, or maybe a specific reason is clearing out the old data just that bit earlier.

Also, I wonder if they should await the result of the action.

Reply:
`afterEach` runs after the transition already started (is confirmed), I think. So it actually runs after `beforeEach` and not when you leave the route.